index.bs

```diff
@@ -1570,70 +1570,63 @@ Observers involve defining a new class, dictionaries for options, and a new type
-  :: Any nonnegative JavaScript number in the integer-representable range, throwing a
-      <code highlight="js">TypeError</code> outside the range and rounding inside of it
-
-  : <code>bigint</code>
-  :: When the possible integer values range more broadly than JavaScript numbers can accurately represent
-
-Additionally, you can combine any of the above with an extra line in your
-algorithm to validate that the number is within the expected domain-specific
-range, and throwing or performing other actions in response. (While it is very
-rarely appropriate to modify author input by taking it modulo 65536, it might be
-appropriate to take it modulo 360, for example.)
-
-A special case of domain-specific validation, which Web IDL already has you
-covered for, is the 0–255 range. This can be written as <code>[EnforceRange]
-octet</code>: any JavaScript number in the range 0–255, throwing a <code
-highlight="js">TypeError</code> outside the range and rounding inside of it.
-(And indeed, if it turns out that the other power-of-two ranges are semantically
-meaningful for your domain, such that you want the modulo or range-checking
-behavior, feel free to use them.)
+  :: Any positive or zero JavaScript number between -2<sup>53</sup> and 2<sup>53</sup>,
+     rounded to the nearest integer.
+     If a number outside this range is given,
+     the generated bindings will throw a <code highlight="js">TypeError</code>.
+
+JavaScript has only one numeric type, `Number`:
+IEEE 754 double-precision floating point, including ±0, ±Infinity, and NaN.
+[[!WEBIDL]] numeric types represent
+rules for modifying any JavaScript number
+to belong to a subset with particular properties.
+These rules are run when a number is passed to the interface defined in IDL,
+whether a method or a property setter.
+
+If you have extra rules which need to be applied to the number,
+you can specify those in your algorithm.
 
 <div class="note">
-Those coming from other languages should carefully note that despite their
-names, <code>long long</code> and <code>unsigned long long</code> only have 53
-bits of precision, and not 64.
+The <a href="https://heycam.github.io/webidl/#es-integer-types">WEBIDL rules</a>
+for converting a JavaScript number to a number with fewer bits,
+such as an `octet` (8 bits, in the range [0, 255]),
+involves taking the modulo of the JavaScript number.
+For example, to convert a JavaScript number value of 300
+to an `octet`,
+the bindings will first compute 300 modulo 255,
+so the resulting number will be 45,
+which might be surprising.
+
+Instead, you can use
+<code><a href="https://heycam.github.io/webidl/#EnforceRange">[EnforceRange]</a> octet</code>
+to throw a `TypeError` for values outside of the `octet` range,
+or <code><a href="https://heycam.github.io/webidl/#Clamp">[Clamp]</a> octet</code>
+to clamp values to the octet range (for example, converting 300 to 255).
+
+This also works for the other shorter types, such as `short` or `long`.
 </div>
 
```

Comment:
```suggestion

<code>bigint</code> should be used only when values greater than 2<sup>53</sup> 
 or less than -2<sup>53</sup> are expected.
 
 An API should not support both `BigInt` and `Number` simultaneously, 
either by supporting both types polymorphously,
or by adding separate, otherwise identical APIs which take `BigInt` and `Number`.
This risks losing precision through implicit conversions, 
which defeats the purpose of `BigInt`.
```